src/main/java/ch/uzh/ifi/hase/soprafs21/security/JwtTokenAuthenticationFilter.java

```diff
@@ -0,0 +1,69 @@
+package ch.uzh.ifi.hase.soprafs21.security;
+
+import ch.uzh.ifi.hase.soprafs21.security.config.SecurityConstants;
+import ch.uzh.ifi.hase.soprafs21.service.JwtTokenUtil;
+import ch.uzh.ifi.hase.soprafs21.service.UserService;
+import io.jsonwebtoken.Claims;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class JwtTokenAuthenticationFilter extends OncePerRequestFilter {
```

Comment:
This code is complex enough and good candidate for automation test

Reply:
Yes good idea.. I will add test for this